examples/smart_contracts/v2/javascript/ContractAccount.js

```diff
@@ -1,78 +1,77 @@
-// const server = "<algod-address>";
-// const port = <algod-port>;
-
-const token = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
+// Create an algod client, using default sandbox parameters here
+const token = "a".repeat(64);
 const server = "http://localhost";
 const port = 4001;
-
-
-// Import the filesystem module 
-const fs = require('fs'); 
-
-// create an algod v2 client
 let algodclient = new algosdk.Algodv2(token, server, port);
 
 (async () => {
+    // Get receiver account info from sandbox
+    const accounts = await getAccounts();
+    const receiverAccount = accounts[0];
+    console.log("Receiver Address: " + receiverAccount.addr);
 
     // get suggested parameters
     let params = await algodclient.getTransactionParams().do();
-    // comment out the next two lines to use suggested fee
-    // params.fee = 1000;
-    // params.flatFee = true;
     console.log(params);
-    // create logic sig
-    // samplearg.teal
-    // This code is meant for learning purposes only
-    // It should not be used in production
-    // arg_0
-    // btoi
-    // int 12345
-    // ==
-    // see more info here: https://developer.algorand.org/docs/features/asc1/sdks/#accessing-teal-program-from-sdks
-    let  fs = require('fs'),
+
+    // Read TEAL file. See more info here: https://developer.algorand.org/docs/features/asc1/sdks/#accessing-teal-program-from-sdks
+    let fs = require('fs'),
         path = require('path'),
         filePath = path.join(__dirname, 'samplearg.teal');
-        // filePath = path.join(__dirname, '<filename>');
     let data = fs.readFileSync(filePath);
+
+    //Compile the program against the algod
     let results = await algodclient.compile(data).do();
     console.log("Hash = " + results.hash);
     console.log("Result = " + results.result);
-    // let program = new Uint8Array(Buffer.from("base64-encoded-program" < PLACEHOLDER >, "base64"));
     let program = new Uint8Array(Buffer.from(results.result, "base64"));
-    //let program = new Uint8Array(Buffer.from("AiABuWAtFyIS", "base64"));
-    // Use this if no args
-    // let lsig = algosdk.makeLogicSig(program);
 
-    // String parameter
-    // let args = ["<my string>"];
-    // let lsig = algosdk.makeLogicSig(program, args);
-    // Integer parameter
-
-    let args = getUint8Int(12345);
+    // Create the lsig account, passing arg
+    let args = getUint8Int(123);
     let lsig = new algosdk.LogicSigAccount(program, args);
-    console.log("lsig : " + lsig.address());   
-    
+    console.log("lsig : " + lsig.address());
+
+    // fund the contract account to test payment from the contract
+    const atc = new algosdk.AtomicTransactionComposer();
+    const signer = algosdk.makeBasicAccountTransactionSigner(receiverAccount)
+    const ptxn = new Transaction({
+        from: receiverAccount.addr,
+        to: lsig.address(),
+        amount: (algosToMicroalgos(0.1) + algosdk.ALGORAND_MIN_TX_FEE + 10000),
```

Comment:
why add the extra at the end? more transactions?

Reply:
do you mean the 10000? That's the amount the contract will send to the receiver. In order to stay above the min balance, I need to send the extra 10000 to cover the payment transaction.